pkg/cloud/onprem/onprem.go

```diff
@@ -59,23 +59,25 @@ func NewOnPremBuiltin(kubeConfigFile string) *starlark.Builtin {
 
 // KubeConfig is part of the cloud.KubernetesVendor interface.
 func (o *OnPrem) KubeConfig(ctx context.Context) (*rest.Config, error) {
-	kubeConfigVaultPath := o.AbstractKubeVendor.AddonSkyCtx(
-		map[string]string{}).Attrs["vaultkubeconfig"].(starlark.String).String()
+	if vaultKubeConfig, ok := o.AbstractKubeVendor.AddonSkyCtx(
+		map[string]string{}).Attrs["vaultkubeconfig"]; ok {
+		kubeConfigVaultPath := vaultKubeConfig.(starlark.String).String()
 
-	// Should only access vault kubeconfig if the kubeConfigFile flag was not set
-	// and the vaultkubeconfig attribute is set in the star config.
-	if len(kubeConfigVaultPath) > 0 && len(o.kubeConfigFile) < 1 {
-		// Remove the surrounding quotes from the Starlark string
-		kubeConfigVaultPath = strings.Trim(kubeConfigVaultPath, `"`)
+		// Should only access vault kubeconfig if the kubeConfigFile flag was not set
```

Comment:
hmm I just notice that this comment is not ture. If the vaultkubeconfig attribute is set then the onprem() cluster will always use vaultkubeconfig rather than kubeConfigFile flag. I am fine with the existing behavior. Since you are on this, perhaps you could update the comment here to reflect the truth?

Reply:
Hmm maybe I'm missing something, but [line 66](https://github.com/cruise-automation/isopod/pull/27/files#diff-8458d1eb0cc345fde5af185bc24ca63bR68) `if len(kubeConfigVaultPath) > 0 && len(o.kubeConfigFile) < 1 {`  only enters the if clause if both conditions are true. So if `kubeConfigVaultPath` is set, but also `o.kubeConfigFile` would not be empty, we wouldn't enter this if clause, even though `kubeConfigVaultPath` is set.
Hence we would fall back to using the file in this case ([line 82](https://github.com/cruise-automation/isopod/pull/27/files#diff-8458d1eb0cc345fde5af185bc24ca63bR82)):
`return clientcmd.BuildConfigFromFlags("", o.kubeConfigFile)` making the comment correct again.
Let me know if I'm getting this wrong :D

Reply:
you are absolutely right